aether/sdk/drf/tests/test_views.py

```diff
@@ -0,0 +1,115 @@
+from aether.sdk.tests.fakeapp import models
+
+TEST_REALM = 'realm-test'
+
+
+class ViewsTest(UrlsTestCase):
+
+    def setUp(self):
+        super(ViewsTest, self).setUp()
+
+        username = 'user'
+        email = 'user@example.com'
+        password = 'secretsecret'
+        user = get_user_model().objects.create_user(username, email, password)
+
+        self.request = RequestFactory().get('/')
+        self.request.COOKIES[settings.REALM_COOKIE] = TEST_REALM
+        self.request.user = user
+
+        self.assertTrue(self.client.login(username=username, password=password))
+        self.client.cookies[settings.REALM_COOKIE] = TEST_REALM
+        test_names = ['one', 'two', 'three']
+        for i in range(3):
+            parent = models.TestModel.objects.create(name=test_names[i])
+            for _ in range(10):
+                models.TestChildModel.objects.create(
+                    name=str(random.randint(1, 100)),
+                    parent=parent,
+                )
+
+    def test_delete_by_filters(self):
+        org_count = models.TestChildModel.objects.count()
+        parent = models.TestModel.objects.first()
+        filtered_count = models.TestChildModel.objects.filter(parent=parent).count()
+        self.assertEqual(org_count, 30)
+        self.assertEqual(filtered_count, 10)
+
+        url = f'{reverse("testchildmodel-filtered-delete")}?search={parent.name}'
+        self.client.delete(url)
+        self.assertEqual(
+            models.TestChildModel.objects.count(),
+            (org_count - filtered_count)
+        )
+        filtered_count = models.TestChildModel.objects.filter(parent=parent).count()
+        self.assertEqual(filtered_count, 0)
+
+    def test_update_by_filters(self):
+        org_count = models.TestChildModel.objects.count()
+        parent = models.TestModel.objects.first()
+        filtered_count = models.TestChildModel.objects.filter(parent=parent).count()
+        self.assertEqual(org_count, 30)
+        self.assertEqual(filtered_count, 10)
+        url = f'{reverse("testchildmodel-filtered-partial-update")}?search={parent.name}'
+        update_fields = {
+            'name': 'new-name',
+        }
+        response = self.client.patch(
+            url,
+            data=update_fields,
+            content_type='application/json'
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        [
+            self.assertEqual(i.name, 'new-name')
+            for i in models.TestChildModel.objects.filter(parent=parent)
+        ]
+
+        update_fields = {
+            'parent': 20,
+        }
+
+        response = self.client.patch(
+            url,
+            data=update_fields,
+            content_type='application/json'
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        [
+            self.assertEqual(i.parent.id, parent.id)
+            for i in models.TestChildModel.objects.filter(parent=parent)
```

Comment:
Does this loop do ... something?

Reply:
Yes to check that the updates wasn't effected... but now that i think of it.. its impossible 🤣

Reply:
Now I feel better _misery loves companion_